### PR TITLE
Add Page Schema for SEO

### PR DESCRIPTION
**Pull request by jestapinski**
This PR looks to complete [DEVHUB-38](https://jira.mongodb.org/browse/DEVHUB-38) by adding stringified schema for pages to the `<head />`.

I'll comment some places in the code I have questions about the schema. I documented my understanding of the schema in `get-page-schema.test.js`.

### CODE REVIEW

File: src/templates/article.js [hunk 1]
```diff
@@ -179,6 +182,17 @@ const Article = props => {
                 <meta property="og:title" content={articleTitle}></meta>
                 <meta property="og:url" content={articleUrl}></meta>
                 <link rel="canonical" href={articleUrl}></link>
+                <script type="application/ld+json">
+                    {getPageSchema(articleUrl, {
+                        articleBody: __refDocMapping.source,
```
Comment (jestapinski) [inline src/templates/article.js]: This is pretty-raw content from the rST, but not sure if there is something better we can use here. 

```
:template: devhub-article\n\n.. ---------------------------------------------------------------\n.. META FIELDS FOR SEO / SOCIAL\n.. ---------------------------------------------------------------\n\n.. meta-description:: \n\n   This post will begin by describing the database capabilities required by modern multi-data center applications.\n\n.. twitter::\n   :site: @mongodb\n   :creator: @jayrunkel\n   :title: Active-Active Application Architectures with MongoDB\n   :image: https://webassets.mongodb.com/_com_assets/cms/image4-3d593nantp.png\n\n   This post will begin by describing the database capabilities required by modern multi-data center applications\n\n.. ---------------------------------------------------------------\n.. ARTICLE METADATA FIELDS (TO POPULATE FILTERS AND ARTICLE PAGE)\n.. ---------------------------------------------------------------\n\n.. include:: /includes/authors/runkel-jay.rst\n\n.. pubdate:: 2017-12-08\n\n.. type:: article\n\n.. level:: advanced\n\n.. tags:: \n\n   * Technical\n\n.. products::\n\n   * MongoDB\n\n.. atf-image:: /images/atf-images/generic/purple.png\n\n.. ---------------------------------------------------------------\n.. ARTICLE CONTENT\n.. ---------------------------------------------------------------\n\n====================================================\nActive-Active Application Architectures with MongoDB\n====================================================\n\n.. introduction::\n\n    Introduction\n    ------------\n\n    Determining the best database for a modern application to be deployed\n    across multiple data centers requires careful evaluation to accommodate\n    a variety of complex application requirements. The database will be\n    responsible for processing reads and writes in multiple geographies,\n    replicating changes among them, and providing the highest possible\n    availability, consistency, and durability guarantees. But not all\n    technology choices... (rest of article)
```

File: src/utils/get-page-schema.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
+export const getPageSchema = (url, articleInfo = null) => {
```
Comment (jestapinski) [inline src/utils/get-page-schema.js]: Does this also need to be added to `/`, `/learn`, `/community` as well?

File: tests/unit/get-page-schema.test.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
+import { getPageSchema } from '../../src/utils/get-page-schema';
+
+it('should correctly generate a schema for SEO on non-article pages', () => {
+    /*
+    Should have a schema looking like the following:
+    {{
+      '@context': 'http://schema.org',
+      '@type': 'Organization',
```
Comment (jestapinski) [inline tests/unit/get-page-schema.test.js]: Is this value always `Organization`?

File: tests/unit/get-page-schema.test.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+it('should correctly generate a schema for SEO on article pages', () => {
+    /*
+  Should have a schema similar to the following:
+  {{'@context': 'https://schema.org',
+    '@type': 'BlogPosting',
+    url: articleURL,
+    headline: articleTitle,
+    description: aritcleDescription,
+    datePublished: datePublished,
+    datemodified: datemodified,
```
Comment (jestapinski) [inline tests/unit/get-page-schema.test.js]: Should this be camelCased? The example in the ticket was not.

Should dates be formatted in a specific way?

File: src/utils/get-page-schema.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    const schema = {
+        '@context': 'http://schema.org',
+        url: url,
+    };
+    const siteImageObject = {
+        '@type': 'imageObject',
+        url:
+            'https://webassets.mongodb.com/_com_assets/cms/mongodb_logo1-76twgcu2dm.png',
```
Comment (jestapinski) [inline src/utils/get-page-schema.js]: I'm assuming this is also the mongo logo to reference here (same as in the blog).

File: src/utils/get-page-schema.js (hunk 2 above)
Comment (jbouzi12) [reply src/utils/get-page-schema.js]: Yea I think we should add these to those pages as well.

File: src/templates/article.js (hunk 1 above)
Comment (greggb) [reply src/templates/article.js]: Not sure, still waiting for the final schema from marketing, so maybe they won't want the whole thing.

File: src/utils/get-page-schema.js (hunk 2 above)
Comment (greggb) [reply src/utils/get-page-schema.js]: Will get confirmation, but I believe the abbreviated version, yes

File: src/utils/get-page-schema.js (hunk 5 above)
Comment (greggb) [reply src/utils/get-page-schema.js]: Yea, you can grab the logo from the ticket and import it in this file

File: tests/unit/get-page-schema.test.js (hunk 3 above)
Comment (greggb) [reply tests/unit/get-page-schema.test.js]: Yes

File: tests/unit/get-page-schema.test.js (hunk 4 above)
Comment (greggb) [reply tests/unit/get-page-schema.test.js]: I'd just follow the spec, but they'll be sending a confirmed markup soon hopefully

https://jsonld.com/blog-post/
https://developers.google.com/search/docs/guides/intro-structured-data

**Comment by greggb** [comment]
Just going to close for now, but we'll revive when we get the final schema

**Pull request closed without merging**